### ISSUE TITLE
iOS 13.3.1 Update (AR Foundation)

### ISSUE THREAD

**Issue by YoungRichi** [Unity-Technologies/arfoundation-samples#415]
Hello,
I have found an issue with iOS 13.3.1. The iPhone with the latest iOS update cannot run an ARFoundation project that i have been working on. However, it runs perfectly fine on another iPhone with iOS 13.3.

[https://forum.unity.com/threads/ar-foundation-unity20193-1f-building-ios.831571/](https://forum.unity.com/threads/ar-foundation-unity20193-1f-building-ios.831571/)
I believe i am not the only person with the issue. 
Any advice on how to troubleshoot?
Thank you.

**Comment by DavidMohrhardt** [comment]
I will respond to the forum post as well (since it appears others are having this issue) but would you mind posting your manifest for the project so we can start with a basic repro project?

**Comment by YoungRichi** [comment]
I have attached more info on the forum. Basically a standard AR foundation set up with all the latest updates. Is there any specific info you need from me?

**Comment by lydiajessup** [comment]
I have the same issue with my iPad that just automatically updated to 13.3.1. - I would really love to figure out a fix!

**Comment by DavidMohrhardt** [comment]
We have tried to reproduce and are unable to do so with the devices on hand.  Can you please submit a bug via the bug tracker and include your project?

Additionally can you include the exact version of XCode you are using to build the application?

https://unity3d.com/unity/qa/bug-reporting

**Comment by DavidMohrhardt** [comment]
Also please let us know when you do, so we can triage it asap.

**Comment by YoungRichi** [comment]
Thank you for writing back.
I have submitted a bug report on Unity which includes my project. 
I am using an iMac 2017 with the latest XCode(11.3.1) and latest macOS (Catalina 10.15.3)

**Comment by tdmowrer** [comment]
> I have submitted a bug report on Unity which includes my project.

Thank you. Can you post the issue number here? (It makes it easier for us to find.)

**Comment by YoungRichi** [comment]
Absolutely. It is Case 1222831.

**Comment by tdmowrer** [comment]
Got it.

**Comment by tdmowrer** [comment]
@YoungRichi The scene `SampleScene` in your project worked fine for me using

Unity 2019.3.2f1
Xcode 11.3.1 (11C505)
iPad Pro MPDY2LL/A

What device and Xcode version are you using?

**Comment by YoungRichi** [comment]
Really? I am very surprised.
I am using Unity 2019.3.2f1, Version 11.3.1 (11C504) on iPhone 8Plus (13.3.1).

**Comment by AbrarAmin1** [comment]
i have same issue with my App.
On previous devices like 7plus 7, even on iphone X it is working but Crashing on XR and XS.
One more thing, live version of the app is working on XR and XS but the test flight is crashing on AR part.

**Comment by tdmowrer** [comment]
Do you have a callstack for the crash?

**Comment by AbrarAmin1** [comment]
> Do you have a callstack for the crash?

do you mean crash report?

**Comment by tdmowrer** [comment]
The crash report should contain a callstack, but the entire crash report could be useful if you are willing to share it.

**Comment by AbrarAmin1** [comment]
[crashlog.txt](https://github.com/Unity-Technologies/arfoundation-samples/files/4255965/crashlog.txt)

**Comment by AbrarAmin1** [comment]
> The crash report should contain a callstack, but the entire crash report could be useful if you are willing to share it.

here is the crash report

**Comment by tdmowrer** [comment]
Are you sure you are using ARFoundation?

```
0   libsystem_platform.dylib      	0x00000001b9b5a594 _platform_memmove + 164
1   Freddy                        	0x0000000103af78ec il2cpp::icalls::mscorlib::System::Runtime::InteropServices::Marshal::copy_from_unmanaged(long, in... + 76 (Marshal.cpp:63)
2   Freddy                        	0x00000001044a682c MarshalDirectionalLightEstimate_MarshalCoefficients_m10B24466A4476CC452A5BD06979008DEA75223A7 + 140 (Assembly-CSharp12.cpp:24665)
3   Freddy                        	0x00000001044ab53c UnityMarshalLightData_op_Implicit_mAB2E438F89C9A7AD564DDF148171BFB91F1D75AF + 132 (Assembly-CSharp12.cpp:24636)
4   Freddy                        	0x00000001044a8d68 UnityARSessionNativeInterface__frame_update_m7E9E0776881076EEAF54F8CE24680FF83E7DDCC4 + 192 (Assembly-CSharp12.cpp:28224)
5   Freddy                        	0x0000000102a0cc28 ReversePInvokeWrapper_UnityARSessionNativeInterface__frame_update_m7E9E0776881076EEAF54F8CE24680F... + 52 (Assembly-CSharp12.cpp:26354)
6   Freddy                        	0x0000000102ab7a70 __41-[UnityARSession session:didUpdateFrame:]_block_invoke + 48 (ARSessionNative.mm:661)
```

Suggests [this callback](https://bitbucket.org/Unity-Technologies/unity-arkit-plugin/src/7793305dc6dd033221f4b514d9b9310bc7cc74e0/Assets/UnityARKitPlugin/Plugins/iOS/UnityARKit/NativeInterface/UnityARSessionNativeInterface.cs#lines-749) from the UnityARKit BitBucket plugin.

**Comment by AbrarAmin1** [comment]
> Are you sure you are using ARFoundation?
> 
> ```
> 0   libsystem_platform.dylib      	0x00000001b9b5a594 _platform_memmove + 164
> 1   Freddy                        	0x0000000103af78ec il2cpp::icalls::mscorlib::System::Runtime::InteropServices::Marshal::copy_from_unmanaged(long, in... + 76 (Marshal.cpp:63)
> 2   Freddy                        	0x00000001044a682c MarshalDirectionalLightEstimate_MarshalCoefficients_m10B24466A4476CC452A5BD06979008DEA75223A7 + 140 (Assembly-CSharp12.cpp:24665)
> 3   Freddy                        	0x00000001044ab53c UnityMarshalLightData_op_Implicit_mAB2E438F89C9A7AD564DDF148171BFB91F1D75AF + 132 (Assembly-CSharp12.cpp:24636)
> 4   Freddy                        	0x00000001044a8d68 UnityARSessionNativeInterface__frame_update_m7E9E0776881076EEAF54F8CE24680FF83E7DDCC4 + 192 (Assembly-CSharp12.cpp:28224)
> 5   Freddy                        	0x0000000102a0cc28 ReversePInvokeWrapper_UnityARSessionNativeInterface__frame_update_m7E9E0776881076EEAF54F8CE24680F... + 52 (Assembly-CSharp12.cpp:26354)
> 6   Freddy                        	0x0000000102ab7a70 __41-[UnityARSession session:didUpdateFrame:]_block_invoke + 48 (ARSessionNative.mm:661)
> ```
> 
> Suggests [this callback](https://bitbucket.org/Unity-Technologies/unity-arkit-plugin/src/7793305dc6dd033221f4b514d9b9310bc7cc74e0/Assets/UnityARKitPlugin/Plugins/iOS/UnityARKit/NativeInterface/UnityARSessionNativeInterface.cs#lines-749) from the UnityARKit BitBucket plugin.

Sorry i didn't mention. i am not using AR Foundation. 
But the issue is same. Isn't it.?

**Comment by tdmowrer** [comment]
ARFoundation does not share a common codebase with UnityARKit, so these seem like to unrelated issues with the same symptom. UnityARKit is no longer maintained, and to fix it in ARFoundation, we would need to reproduce the problem or at least have a crash report to even begin to debug the issue.

**Comment by AbrarAmin1** [comment]
> ARFoundation does not share a common codebase with UnityARKit, so these seem like two unrelated issues with the same symptom. UnityARKit is no longer maintained, and to fix it in ARFoundation, we would need to reproduce the problem or at least have a crash report to even begin to debug the issue.

ok. Thanks for the help. i will look for the related thread

**Comment by tdmowrer** [comment]
@YoungRichi 

> Really? I am very surprised.
> I am using Unity 2019.3.2f1, Xcode Version 11.3.1 (11C504) on iPhone 8Plus (13.3.1).
> It is working on an older iPhone 7 (13.3) for me
> 
> https://www.youtube.com/watch?v=OVS6eLFNd1I
> Out of curious, I upgraded my personal iPad Pro 2017 to iOS 13.3.1. I was able to reproduce the same error. How could that be?

Impossible to say without more information. Can you provide a callstack for the crash?

**Comment by YoungRichi** [comment]
I believe I have found the reason why you cant reproduce the error. The console output i get from XCode is exactly the same as this user.
[https://answers.unity.com/questions/1696331/ios-builds-in-xcode-with-no-errors-but-crashes-upo.html](https://answers.unity.com/questions/1696331/ios-builds-in-xcode-with-no-errors-but-crashes-upo.html)
It appears that it only affect users with free Apple IDs. 
I also want to point out that there is no crash log because it builds successfully on Xcode. It only then crashes on the device. 
The debugger wont be able to hit any breakpoint because it builds successfully on Xcode.
I have tried to manually set a breakpoint on main but the thread is not giving me any meaningful information. Can you advice me on where to retrieve the callstack in this case?

**Comment by tdmowrer** [comment]
> It appears that it only affect users with free Apple IDs.

Okay, cool. That sounds like a new rule/change by Apple that affects all apps, not just Unity.

> I also want to point out that there is no crash log because it builds successfully on Xcode. It only then crashes on the device.

But you should be able to get a callstack from Xcode while debugging your app on device, like in this screenshot:
<img width="355" alt="Screen Shot 2020-02-26 at 10 58 29 AM" src="https://user-images.githubusercontent.com/4934281/75377658-2141a180-5887-11ea-9da2-450e028dfe80.png">

**Comment by YoungRichi** [comment]
>> But you should be able to get a callstack from Xcode while debugging your app on device, like in this 

Please excuse my ignorance. Where did you set your breakpoint?  
I put a breakpoint on main. 
<img width="661" alt="Screen Shot 2020-02-26 at 4 58 07 PM" src="https://user-images.githubusercontent.com/34274756/75392237-94fca200-58b9-11ea-9c4e-db56baf0df6d.png">
and this is what i get..
<img width="460" alt="Screen Shot 2020-02-26 at 4 58 34 PM" src="https://user-images.githubusercontent.com/34274756/75392254-9f1ea080-58b9-11ea-8379-83612c395b9b.png">

Without the breakpoint, Xcode compiles the project and quits.

**Comment by tdmowrer** [comment]
I only added a breakpoint so that I could show where the callstack appears. Generally, when there is a crash on device, the program halts and Xcode highlights the line with the problem. At that point you can see the callstack. You're saying your app simply exists cleanly? If so, that doesn't sound like a crash. Are there any messages in the console window?

**Comment by YoungRichi** [comment]
I see what you mean. You are correct. I should have mentoined it earlier. It exits cleanly. With the message: "Finishing running ProjectName on iPhone"
![Screen Shot 2020-02-26 at 10 56 34 PM](https://user-images.githubusercontent.com/34274756/75410863-3ea85700-58eb-11ea-8a2d-fb7c648ce2ea.png)
The console log shows this (Similar to this person from this Question) [https://answers.unity.com/questions/1696331/ios-builds-in-xcode-with-no-errors-but-crashes-upo.html](https://answers.unity.com/questions/1696331/ios-builds-in-xcode-with-no-errors-but-crashes-upo.html) 

<img width="710" alt="Screen Shot 2020-02-21 at 9 56 08 PM" src="https://user-images.githubusercontent.com/34274756/75410949-9777ef80-58eb-11ea-95a9-c72cf05261be.png">

**Comment by tdmowrer** [comment]
Ah, I see the "code signature invalid" error message. There's an [Apple forum thread](https://forums.developer.apple.com/thread/128925) about this -- it looks like it affects any project that involves an embedded framework. I don't know whether Apple consider this a bug. 

While this isn't a Unity-specific issue, there was a change in 2019.3 where Unity uses an embedded framework where it didn't in 2019.2. A temporary workaround would be to downgrade to Unity 2019.2.

**Comment by YoungRichi** [comment]
I see. Thank you for letting us know.

**Comment by tdmowrer** [comment]
Marking as closed as this is not actionable by Unity.

**Comment by tdmowrer** [comment]
This appears to have been resolved in iOS 13.4.1
